src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/transcripts-list-item-info/TranscriptsListItemInfo.tsx

```diff
@@ -0,0 +1,161 @@
+  const { transcript } = props;
+
+  const getTranscriptLocation = () => {
+    const { start, end } = getFeatureCoordinates(transcript);
+    const startStr = getCommaSeparatedNumber(start);
+    const endStr = getCommaSeparatedNumber(end);
+    const regionName = getRegionName(transcript);
+    return `${regionName}:${startStr}-${endStr}`;
+  };
+
+  const getRNALength = () => {
+    const rnaLength = transcript.exons.reduce((length, exon) => {
+      const { start, end } = getFeatureCoordinates(exon);
+      return length + (end - start + 1);
+    }, 0);
+
+    return getCommaSeparatedNumber(rnaLength);
+  };
+
+  const getFirstAndLastCodingExonIndexes = () => {
+    const { exons, cds } = transcript;
+    let firstCodingExonIndex = 0;
+    let lastCodingExonIndex = exons.length - 1;
+
+    if (cds) {
+      for (let index = 0; index < exons.length; index += 1) {
+        const exonLocation = getFeatureCoordinates(exons[index]);
+
+        if (exonLocation.start <= cds.start && exonLocation.end >= cds.start) {
+          firstCodingExonIndex = index;
+          break;
+        }
+      }
+
+      for (let index = exons.length - 1; index > -1; index -= 1) {
+        const exonLocation = getFeatureCoordinates(exons[index]);
+
+        if (exonLocation.start <= cds.end && exonLocation.end >= cds.end) {
+          lastCodingExonIndex = index;
+          break;
+        }
+      }
+    }
+
+    return {
+      firstCodingExonIndex,
+      lastCodingExonIndex
+    };
+  };
+
+  // FIXME: remove this when the amino acid length can be retrieved via the API
+  const getAminoAcidLength = () => {
+    const { exons, cds } = transcript;
+
+    if (cds) {
+      const {
+        firstCodingExonIndex,
+        lastCodingExonIndex
+      } = getFirstAndLastCodingExonIndexes();
+      let cdsLength = 0;
+
+      // add coding length of the first coding exon
+      const { end: firstCodingExonEnd } = getFeatureCoordinates(
+        exons[firstCodingExonIndex]
+      );
+      cdsLength += firstCodingExonEnd - cds.start + 1;
+
+      // add coding length of the last coding exon
+      const { start: lastCodingExonStart } = getFeatureCoordinates(
+        exons[lastCodingExonIndex]
+      );
+      cdsLength += cds.end - lastCodingExonStart + 1;
+
+      // add coding length of exons between first and last coding exons
+      for (
+        let index = firstCodingExonIndex + 1;
+        index <= lastCodingExonIndex - 1;
+        index += 1
+      ) {
+        const { start, end } = getFeatureCoordinates(exons[index]);
+        cdsLength += end - start + 1;
+      }
+
+      const aminoAcidLength = Math.floor(cdsLength / 3);
+      return aminoAcidLength;
+    } else {
+      return 0;
+    }
+  };
+
+  const getNumberOfCodingExons = () => {
+    const {
+      firstCodingExonIndex,
+      lastCodingExonIndex
+    } = getFirstAndLastCodingExonIndexes();
+    return getCommaSeparatedNumber(
+      lastCodingExonIndex - firstCodingExonIndex + 1
```

Comment:
Why the +1?

Reply:
Because this calculation would give the difference rather than all the coding exons without padding with 1. For example:

```
All exons = 27
First coding exon = 1
Last coding exon = 26
Coding difference = Last coding exon - First coding exon = 25
```

Reply:
I am confused. Why does first coding exon have an index of 1? I thought it will have an index of 0, as follows from this for-loop:

```
      for (let index = 0; index < exons.length; index += 1) {
        const exonLocation = getFeatureCoordinates(exons[index]);

        if (exonLocation.start <= cds.start && exonLocation.end >= cds.start) {
          firstCodingExonIndex = index;
          break;
        }
      }
```

Reply:
Sorry, I now see that you are right.